applications/CoSimulationApplication/python_scripts/solver_wrappers/external/acusolve_wrapper.py

```diff
@@ -71,121 +73,181 @@ def __init__(self, settings, model, solver_name):
+        #   2. Solve the step
+        #   3. Export the temperature value to Kratos
+        #   4. Wait for and Import the heat sources from Kratos
+        # The process starts when the AdvanceInTime or RepeatTimeStep signal is sent to Acusolve
+        start_time = time.time()
+        print("{} :: SolveSolutionStep starts...".format(self.name))
+
+        # In debug mode
+        if self.debug_mode :
+            print("{} :: Exporting current I/O data to VTK output".format(self.name))
+            self.vtk_output.PrintOutput()
+            self.vtk_output.ExecuteFinalizeSolutionStep()
+
+        # Start the solving process in Acusolve
+        if self.advance_in_time : 
+            print("{} :: Activate next step".format(self.name))
+            self.__SendControlSignal("AdvanceInTime")
+            self.advance_in_time = False
+        else:    
+            print("{} :: Repeat step".format(self.name))
+            self.__SendControlSignal("RepeatTimeStep")
+
+        # Wait for and import the temperature values from Acusolve
+        print("{} :: Import temperature from Acusolve".format(self.name))
         for data_name in self.settings["solver_wrapper_settings"]["import_data"].GetStringArray():
             data_config = { 
                 "type" : "coupling_interface_data",
                 "interface_data" : self.GetInterfaceData(data_name)
             }
             self.ImportData(data_config)
 
+        # Export heat sources to Acusolve (who is waiting for them)
+        print("{} :: Export heat sources to Acusolve".format(self.name))
+        for data_name in self.settings["solver_wrapper_settings"]["export_data"].GetStringArray():
+            data_config = { 
+                "type" : "coupling_interface_data",
+                "interface_data" : self.GetInterfaceData(data_name)
+            }
+            self.ExportData(data_config)
+
+        print("{} :: SolveSolutionStep executed. :: {} s.".format(self.name, time.time() - start_time))
+
+        return
+
+    def Finalize(self):
+        # Inform Acusolve that the overall process is finished.
+        # The order will activated finalizing process in Acusolve.
+        print("{} :: Finalize".format(self.name))
+        print("control signal : exit")
+        self.__SendControlSignal("exit")
+        # If the VTK output process is running, finalize it
+        if self.debug_mode:
+            print("{} :: Finalizing VTK output process".format(self.name))
+            self.vtk_output.ExecuteFinalize()
+        super().Finalize()
 
     def _GetIOType(self):
         return self.settings["io_settings"]["type"].GetString()
 
     def __SendControlSignal(self, signal, settings=None):
+        # The control signal process is used to 
+        # inform Acusolve what to do next, it must be :
+        #   - RepeatTimeStep
+        #   - AdvanceInTime
+        #   - exit
+        print("{} :: Sending signal : {}".format(self.name,signal))
```

Comment:
```suggestion
        print(f"{self.name} :: Sending signal : {signal}")
```

Same for other formatted strings.